Userland/Libraries/LibWeb/Page/Frame.h

```diff
@@ -101,6 +103,12 @@ class Frame : public TreeNode<Frame> {
 
     void did_edit(Badge<EditEventHandler>);
 
+    void register_frame_nesting(URL url);
+    bool is_frame_nesting_allowed(URL url);
+
+    void set_frame_nesting_levels(const HashMap<URL, size_t> frame_nesting_levels) { m_frame_nesting_levels = frame_nesting_levels; };
+    HashMap<URL, size_t> const frame_nesting_levels() { return m_frame_nesting_levels; }
```

Comment:
Let's return this by const-reference to avoid a copy on each call.
Also let's make it const-qualified:
```cpp
HashMap<URL, size_t> const& frame_nesting_levels() const { return m_frame_nesting_levels; }
```